addon/model.js

```diff
@@ -290,12 +289,17 @@ export default class MegamorphicModel extends EmberObject {
   }
 
   rollbackAttributes() {
+    if (DEBUG) {
+      assertNoChanges(this._store);
+    }
+
     let dirtyKeys = recordDataFor(this).rollbackAttributes();
     this._updateCurrentState(loadedSaved);
 
     if (dirtyKeys && dirtyKeys.length > 0) {
       this._notifyProperties(dirtyKeys);
     }
+    flushChanges(this._store);
```

Comment:
If we start with 0 changes, and don't push, why do we have to flush?

Reply:
because `_notifyProperties` goes through the same codepath that defers change notifications

Reply:
see [notifyPropertyChange](https://github.com/hjdivad/ember-m3/blob/c94be430117e7b19dac6ab33018ea609fe7935a9/addon/model.js#L193-L226)